client/finality-grandpa/src/until_imported.rs

```diff
@@ -137,14 +137,16 @@ impl Drop for Metrics {
 	}
 }
 
-/// Buffering imported messages until blocks with given hashes are imported.
-#[pin_project::pin_project]
-pub(crate) struct UntilImported<Block: BlockT, BlockStatus, BlockSyncRequester, I, M: BlockUntilImported<Block>> {
+/// Buffering incoming messages until blocks with given hashes are imported.
+pub(crate) struct UntilImported<Block, BlockStatus, BlockSyncRequester, I, M> where
+	Block: BlockT,
+	I: Stream<Item = M::Blocked> + Unpin,
+	M: BlockUntilImported<Block>,
+{
 	import_notifications: Fuse<TracingUnboundedReceiver<BlockImportNotification<Block>>>,
 	block_sync_requester: BlockSyncRequester,
 	status_check: BlockStatus,
-	#[pin]
-	inner: Fuse<I>,
+	incoming_messages: Fuse<I>,
```

Comment:
On the side this pull requests renames `inner` to `incoming_messages`. I find this name more descriptive compared to the abstract term `inner`. Let me know if you would like me to exclude this change from the pull request.